tests/zkevm/test_worst_compute.py

```diff
@@ -741,6 +741,105 @@ def test_worst_unop(
     )
 
 
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("key_mut", [True, False])
+@pytest.mark.parametrize("val_mut", [True, False])
+def test_worst_tload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    key_mut: bool,
+    val_mut: bool,
+):
+    """Test running a block with as many TLOAD calls as possible."""
+    env = Environment()
+
+    start_key = 41
+    start_val = 43
+    code_prefix = Op.PUSH1(start_val) + Op.PUSH1(start_key) + Op.JUMPDEST
+
+    # If `key_mut` is True, we mutate the key on every iteration of the big loop by mutiplying by
+    # a constant that should produce a long chain of distinct values.
+    code_key_mut = Op.PUSH32(2**256 - 5) + Op.MUL if key_mut else Bytecode()
+
+    # If `val_mut` is True, we mutate the value that will be returned by the nex iteration of
+    # TLOADs.
+    code_val_mut = (
+        Op.DUP2  # MUL current value
+        + Op.PUSH32(2**256 - 5)  # MUL constant factor
+        + Op.MUL  # Calculate new value
+        + Op.SWAP2  # Save it
+        + Op.POP  # Cleanup
+        + Op.TSTORE(Op.DUP2, Op.DUP2)
```

Comment:
I prefer to do linear `+` instead of `opcode(..., ...)` in cases where `Op.DUPX` is used since feels more clear what's happening in the stack.